lib/vm_service_wrapper.dart

```diff
@@ -0,0 +1,268 @@
+
+  @override
+  Stream<Event> get onServiceEvent => _vmService.onServiceEvent;
+
+  @override
+  Stream<Event> get onStderrEvent => _vmService.onStderrEvent;
+
+  @override
+  Stream<Event> get onStdoutEvent => _vmService.onStdoutEvent;
+
+  @override
+  Stream<Event> get onVMEvent => _vmService.onVMEvent;
+
+  @override
+  Future<Success> pause(String isolateId) =>
+      _trackFuture(_vmService.pause(isolateId));
+
+  @override
+  Future<Success> registerService(String service, String alias) =>
+      _trackFuture(_vmService.registerService(service, alias));
+
+  @override
+  void registerServiceCallback(String service, ServiceCallback cb) =>
+      _vmService.registerServiceCallback(service, cb);
+
+  @override
+  Future<ReloadReport> reloadSources(String isolateId,
+          {bool force, bool pause, String rootLibUri, String packagesUri}) =>
+      _trackFuture(_vmService.reloadSources(isolateId,
+          force: force,
+          pause: pause,
+          rootLibUri: rootLibUri,
+          packagesUri: packagesUri));
+
+  @override
+  Future<Success> removeBreakpoint(String isolateId, String breakpointId) =>
+      _trackFuture(_vmService.removeBreakpoint(isolateId, breakpointId));
+
+  @override
+  Future<Success> requestHeapSnapshot(
+          String isolateId, String roots, bool collectGarbage) =>
+      _trackFuture(
+          _vmService.requestHeapSnapshot(isolateId, roots, collectGarbage));
+
+  @override
+  Future<Success> resume(String isolateId, {String step, int frameIndex}) =>
+      _trackFuture(
+          _vmService.resume(isolateId, step: step, frameIndex: frameIndex));
+
+  @override
+  Future<Success> setExceptionPauseMode(String isolateId, String mode) =>
+      _trackFuture(_vmService.setExceptionPauseMode(isolateId, mode));
+
+  @override
+  Future<Success> setFlag(String name, String value) =>
+      _trackFuture(_vmService.setFlag(name, value));
+
+  @override
+  Future<Success> setLibraryDebuggable(
+          String isolateId, String libraryId, bool isDebuggable) =>
+      _trackFuture(
+          _vmService.setLibraryDebuggable(isolateId, libraryId, isDebuggable));
+
+  @override
+  Future<Success> setName(String isolateId, String name) =>
+      _trackFuture(_vmService.setName(isolateId, name));
+
+  @override
+  Future<Success> setVMName(String name) =>
+      _trackFuture(_vmService.setVMName(name));
+
+  @override
+  Future<Success> setVMTimelineFlags(List<String> recordedStreams) =>
+      _trackFuture(_vmService.setVMTimelineFlags(recordedStreams));
+
+  @override
+  Future<Success> streamCancel(String streamId) {
+    _activeStreams.remove(streamId);
+    return _trackFuture(_vmService.streamCancel(streamId));
+  }
+
+  @override
+  Future<Success> streamListen(String streamId) {
+    Future<Success> future;
+    if (!_activeStreams.contains(streamId)) {
+      future = _trackFuture(_vmService.streamListen(streamId));
+    }
+    _activeStreams.add(streamId);
+    return future;
```

Comment:
don't return null, return a future that is completed to what the future when the stream was first listened to instead. This will mean making _activeStreams a map instead of a set.

Reply:
Done.